providers/configcat/src/main/java/dev/openfeature/contrib/providers/configcat/ConfigCatProvider.java

```diff
@@ -0,0 +1,205 @@
+package dev.openfeature.contrib.providers.configcat;
+
+import com.configcat.ConfigCatClient;
+import com.configcat.EvaluationDetails;
+import com.configcat.User;
+import dev.openfeature.sdk.EvaluationContext;
+import dev.openfeature.sdk.EventProvider;
+import dev.openfeature.sdk.Metadata;
+import dev.openfeature.sdk.ProviderEvaluation;
+import dev.openfeature.sdk.ProviderEventDetails;
+import dev.openfeature.sdk.ProviderState;
+import dev.openfeature.sdk.Value;
+import dev.openfeature.sdk.exceptions.GeneralError;
+import dev.openfeature.sdk.exceptions.ProviderNotReadyError;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Provider implementation for ConfigCat.
+ */
+@Slf4j
+public class ConfigCatProvider extends EventProvider {
+
+    @Getter
+    private static final String NAME = "ConfigCat";
+
+    public static final String PROVIDER_NOT_YET_INITIALIZED = "provider not yet initialized";
+    public static final String UNKNOWN_ERROR = "unknown error";
+
+    private ConfigCatProviderConfig configCatProviderConfig;
+
+    @Getter
+    private ConfigCatClient configCatClient;
+
+    @Getter
+    private ProviderState state = ProviderState.NOT_READY;
+
+    private AtomicBoolean isInitialized = new AtomicBoolean(false);
+
+    /**
+     * Constructor.
+     * @param configCatProviderConfig configCatProvider Config
+     */
+    public ConfigCatProvider(ConfigCatProviderConfig configCatProviderConfig) {
+        this.configCatProviderConfig = configCatProviderConfig;
+    }
+
+    /**
+     * Initialize the provider.
+     * @param evaluationContext evaluation context
+     * @throws Exception on error
+     */
+    @Override
+    public void initialize(EvaluationContext evaluationContext) throws Exception {
+        boolean initialized = isInitialized.getAndSet(true);
```

Comment:
The ConfigCatClient.get handles concurrency and multiple initializations. If you call the ConfigCatClient.get multiple times with the same SDK key, it will return a singleton instance of the configcatclient for that SDK Key.  
This is only valid if the hooks will be initialized in the configCatClient.Get call instead of adding them in separate calls (my next comment).
I think this initialized part is unnecessary because of the singleton pattern used in the ConfigCatClient.get, but it is just my opinion.